src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -743,6 +745,14 @@ public static bool TryCompleteSendTo(SafeSocketHandle socket, ReadOnlySpan<byte>
                     errorCode = SocketError.Success;
                     return true;
                 }
+
+                if (socket.SendTimeout > 0 && (Environment.TickCount64 - start) >= socket.SendTimeout)
+                {
+                    // Timeout is set and reached it while trying to send.
```

Comment:
We need a better comment here, explaining that while we can't easily do a strict enforcement of the timeout, this will at least prevent us from exceeding it by too much.